shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java

```diff
@@ -668,16 +666,41 @@ public long configureForTextureLayerComposition(
     return textureId;
   }
 
+  /**
+   * Translates an original touch event to have the same locations as the ones that Flutter
+   * calculates (because original + flutter's - original = flutter's).
+   *
+   * @param originalEvent The saved original input event.
+   * @param pointerCoords The coordinates that Flutter thinks the touch is happening at.
+   */
+  private void translateNonVirtualDisplayMotionEvent(
+      MotionEvent originalEvent, PointerCoords[] pointerCoords) {
+    if (pointerCoords.length < 1) {
+      return;
+    }
+
+    float xOffset = pointerCoords[0].x - originalEvent.getX();
+    float yOffset = pointerCoords[0].y - originalEvent.getY();
+
+    originalEvent.offsetLocation(xOffset, yOffset);
+  }
+
```

Comment:
Could we add a roboelectric test programmatically invokes translateNonVirtualDisplayMotionEvent and toMotionEvent and checks their return values?

Reply:
I wrote a test that invokes toMotionEvent in both the vd and non-vd case, and expects that they have equivalent x and y coords. This isn't exactly what you asked for, but does it make sense to you?